contracts/smart-wallet/managed/ManagedAccount.sol

```diff
@@ -20,10 +20,12 @@ import "lib/dynamic-contracts/src/core/Router.sol";
 contract ManagedAccount is AccountCore, Router {
     address public factory;
 
-    constructor(IEntryPoint _entrypoint) AccountCore(_entrypoint) {}
+    constructor(IEntryPoint _entrypoint) AccountCore(_entrypoint) {
+        _disableInitializers();
```

Comment:
what does this do? is to prevent initializing the implementation itself?

isnt that already handled by the `initializer` modifier?

Reply:
The `_disableInitializer` function has a different job than the `initializer` modifier.

You call the `_disableInitializer` function inside the constructor of the _implementation_ smart contract instance to make sure that no one can call the `initialize` function on the _implementation_ and mutate its state.

From reading the code -- calling `_disableInitializer` inside the implementation contract's constructor is a good practice and a recommendation in the code comments, but not doing this does not affect the proxy contracts that point at it.

Code comments for `_disableInitializer`:
```solidity
    /**
     * @dev Locks the contract, preventing any future reinitialization. This cannot be part of an initializer call.
     * Calling this in the constructor of a contract will prevent that contract from being initialized or reinitialized
     * to any version. It is recommended to use this to lock implementation contracts that are designed to be called
     * through proxies.
     */
```